packages/server/lib/browsers/electron.ts

```diff
@@ -307,7 +307,10 @@ export = {
         cdpSocketServer?.attachCDPClient(cdpAutomation),
         videoApi && recordVideo(cdpAutomation, videoApi),
         this._handleDownloads(win, options.downloadsFolder, automation),
-        await utils.handleDownloadLinksViaCDP(pageCriClient, automation),
+        utils.handleDownloadLinksViaCDP(pageCriClient, automation),
+        // Ensure to clear browser state in between runs. This is handled differently in browsers when we launch new tabs, but we don't have that concept in electron
+        pageCriClient.send('Storage.clearDataForOrigin', { origin: '*', storageTypes: 'all' }),
+        pageCriClient.send('Network.clearBrowserCache'),
```

Comment:
So were we not clearing session/local storage in electron between specs? Is there a test for this?

Reply:
IndexedDB and service workers were added to this test: https://github.com/cypress-io/cypress/pull/28060/files#diff-218614ae013c9499068762e74b5ae5c29ad34236f32ffc71fa73fbd745815a3cR20-R37 which demonstrates the problem